pkg/nodediscovery/nodediscovery.go

```diff
@@ -302,6 +303,16 @@ func (n *NodeDiscovery) UpdateLocalNode() {
 	n.updateLocalNode()
 }
 
+// LocalNode syncs the localNode object with the information stored in the node
+// package and then returns a copy of the localNode object
+func (n *NodeDiscovery) LocalNode() types.Node {
+	n.localNodeLock.Lock()
+	defer n.localNodeLock.Unlock()
+
+	n.fillLocalNode()
+	return n.localNode
```

Comment:
I didn't notice this in the upstream PR, but don't we want to `DeepCopy()` here? `types.Node` contains a few slices that will be aliased. They are modified after `StartDiscovery()` in v1.10/v1.11, but they are in v1.12.

Reply:
Makes sense, in that case yes, we should switch to `DeepCopy()` :+1: 

I'll fill the master PR and backport the fix here and in the v1.10 one